spyder/plugins/editor/plugin.py

```diff
@@ -1796,9 +1796,15 @@ def new(self, fname=None, editorstack=None, text=None):
                         # to this number if there's other untitled file in
                         # spyder. Please see spyder-ide/spyder#7831
                         fname_data = osp.splitext(current_filename)
-                        act_num = int(fname_data[0].split(_("untitled"))[-1])
-                        self.untitled_num = act_num + 1
-
+                        try:
+                            act_num = int(
+                                fname_data[0].split(_("untitled"))[-1])
+                            self.untitled_num = act_num + 1
+                        except ValueError:
+                            # Catch the error in case the user has something
+                            # different from a number after the untitled
+                            # Please see spyder-ide/spyder#12892

```

Comment:
Let's improve this a little bit, like this

```python
# Catch the error in case the user has something
# different from a number after the untitled
# part.
# Please see spyder-ide/spyder#12892
```